stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/logic/channel/internal/ChannelLogic.kt

```diff
@@ -143,6 +144,10 @@ internal class ChannelLogic(
     }
 
     override suspend fun onQueryChannelRequest(channelType: String, channelId: String, request: QueryChannelRequest) {
+        val isChannelMuted = globalMutableState.channelMutes.value.any { it.channel.cid == cid }
+        StreamLog.d(TAG) { "[onQueryChannelRequest] isChannelMuted: $isChannelMuted, cid: $cid" }
+        mutableState._muted.value = isChannelMuted
```

Comment:
This one is to make the following snippet workable:
```kotlin
viewModelScope.launch {
    chatClient.watchChannelAsState(cid, 30).collect { channelState ->
        launch {
            channelState?.muted?.collect { isMuted ->
                // ...
            }
        }
    }
}
```

Reply:
now `isMuted` reflected a real muted state

Reply:
In the example above, why don't we just map instead of launching a coroutine that collects and then launches something else?

```
chatClient.watchChannelAsState(cid, 30).map { channelState ->
  channelState?.muted
}.collect { isMuted -> }
```

I imagine it's because we have other things to do with the channel state, but we could consider not launching to launch on each emission, as that could produce race conditions and CPU overhead